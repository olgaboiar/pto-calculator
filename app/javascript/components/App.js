import React from 'react'
import Description from './Description'
import Greeting from './Greeting'

class App extends React.Component {
  render () {
    return (
      <div>
        <Greeting />
        <Description />
      </div>
    )
  }
}

export default App
