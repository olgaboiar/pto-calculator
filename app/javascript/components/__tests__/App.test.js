import { mount } from "enzyme";
import * as React from "react";
import App from "../App"

describe("App.js", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <App />, { attachToDocument: true }
    );
  
    expect(wrapper).toMatchSnapshot();
  });

  it("contains greeting and description", () => {  
    const wrapper = mount(
      <App />,
    );
  
    const description = wrapper.find('[data-testid="description"]');
    const greeting = wrapper.find('[data-testid="greeting"]');
    expect(description).toHaveLength(1);
    expect(greeting).toHaveLength(1);
  });
})
