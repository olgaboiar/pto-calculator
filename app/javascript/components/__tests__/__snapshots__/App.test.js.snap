// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "data-testid": "greeting",
        },
        "children": Array [
          Object {
            "data": "Welcome to PTO Calculator",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "data-testid": "description",
          },
          "children": Array [
            Object {
              "data": "The only tool that brings your vacation closer!",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-testid": undefined,
          },
          "x-attribsPrefix": Object {
            "data-testid": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-testid": undefined,
        },
        "x-attribsPrefix": Object {
          "data-testid": undefined,
        },
      },
      Object {
        "attribs": Object {
          "data-testid": "description",
        },
        "children": Array [
          Object {
            "data": "The only tool that brings your vacation closer!",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "data-testid": "greeting",
          },
          "children": Array [
            Object {
              "data": "Welcome to PTO Calculator",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "data-testid": undefined,
          },
          "x-attribsPrefix": Object {
            "data-testid": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "data-testid": undefined,
        },
        "x-attribsPrefix": Object {
          "data-testid": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
